aten/src/ATen/TensorOptions.h

```diff
@@ -205,4 +220,28 @@ inline TensorOptions device_index(int32_t device_index) {
 inline TensorOptions requires_grad(bool requires_grad = true) {
   return TensorOptions().requires_grad(requires_grad);
 }
+
+// From Tensor.h:
+
+inline Tensor Tensor::to(TensorOptions options, bool non_blocking) {
+  DeviceGuard guard(options.device());
+  auto copy = options.type().copy(*this, non_blocking);
```

Comment:
The python `.to` doesn't copy if the targeted dtype/device is the same as the ones for input tensor. Does this also do that?

Reply:
Looks like this copies (code for Type::copy):
```
    Tensor r = this->tensor(src.sizes());
    r.copy_(src, non_blocking);
    return r;
```

Reply:
This is something I should fix, thanks